components/binary_sensor/xiaomi.py

```diff
@@ -57,25 +65,29 @@ def is_on(self):
         """Return true if sensor is on."""
         return self._state
 
+    @property
+    def device_state_attributes(self):
+        attrs = {ATTR_NO_MOTION_SINCE: self._no_motion_since}
+        attrs.update(super().device_state_attributes)
+        return attrs
+
     def parse_data(self, data):
-        if 'no_motion' in data:
-            if self._state:
-                self._state = False
-                return True
-            else:
-                return False
+        if NO_MOTION in data:  # handle push from the hub
+            self._no_motion_since = data[NO_MOTION]

```

Comment:
_device_state_attributes is private and thus not available, that's actually the first thing I did. I was thinking how to expose it nicely, but thought that this would do until a nicer way is found.

For the second one, this branch will only be taken when being pushed from the hub, in which case the state is always False -> we have to update anycase (as either the open since has changed, or we may have missed an earlier broadcast&are in incorrect state). What do you think?